core/src/main/scala/io/snappydata/SnappyTableStatsProviderService.scala

```diff
@@ -201,26 +201,49 @@ object SnappyEmbeddedTableStatsProviderService extends TableStatsProviderService
       }
     }
 
-    try {
-      // External Tables
-      val hiveTables: java.util.List[ExternalTableMetaData] =
-        Misc.getMemStore.getExternalCatalog.getHiveTables(true)
-      externalTables = hiveTables.asScala.collect {
-        case table if table.tableType.equalsIgnoreCase("EXTERNAL") =>
-          new SnappyExternalTableStats(table.entityName, table.tableType, table.shortProvider,
-            table.externalStore, table.dataSourcePath, table.driverClass)
+    val hiveTables = Misc.getMemStore.getExternalCatalog.getHiveTables(true).asScala
+
+    val externalTables: mutable.Buffer[SnappyExternalTableStats] = {
+      try {
+        // External Tables
+        hiveTables.collect {
+          case table if table.tableType.equalsIgnoreCase("EXTERNAL") => {
+            new SnappyExternalTableStats(table.entityName, table.tableType, table.shortProvider,
+              table.externalStore, table.dataSourcePath, table.driverClass)
+          }
+        }
       }
-    }
-    catch {
-      case NonFatal(e) => {
-        log.warn("Exception occurred while collecting External Table Statistics: "
-            + e.getMessage, e)
+      catch {
+        case NonFatal(e) => {
+          log.warn("Exception occurred while collecting External Table Statistics: "
+              + e.getMessage, e)
+          mutable.Buffer.empty[SnappyExternalTableStats]
+        }
       }
     }
 
     if (resultObtained) {
+      // Return updated tableSizeInfo
+      // Map to hold hive table type against table names as keys
+      val tableTypesMap: mutable.HashMap[String, String] = mutable.HashMap.empty[String, String]
+      hiveTables.foreach(ht => {
+        val key = ht.schema.toString.concat("." + ht.entityName)
+        tableTypesMap.put(key.toUpperCase, ht.tableType)
```

Comment:
@snappy-sachin This is an inconsistency in ExternalTableMetaData that is sometimes holding the StructType object in schema field with full name in entityName (for getHiveTableMetaData) while for getHiveTables it is returning the schema name of table with just the name in entityName. Please file a separate issue to correct this inconsistency.